packages/core/src/ObservableActorRef.ts

```diff
@@ -74,12 +73,19 @@ export class ObservableActorRef<TEvent extends EventObject, TSnapshot>
         ? (event as LifecycleSignal)
         : toSCXMLEvent(event as TEvent);
 
+    const previous = this.current;
+
     this.current = this.behavior.transition(
       this.current,
       typeof this.context._event.type !== 'string'
         ? (this.context._event as LifecycleSignal)
         : (this.context._event as SCXML.Event<TEvent>).data,
       this.context
     );
+
+    if (previous !== this.current) {
+      const current = this.current;
+      this._observers.forEach((observer) => observer.next?.(current));
+    }
```

Comment:
This was a crucial missing piece. You could have subscribed to the ref - but u wouldn't ever be notified about any change.